runelite-client/src/main/java/net/runelite/client/plugins/twitch/TwitchPlugin.java

```diff
@@ -0,0 +1,418 @@
+		}
+
+		//Show the message in game with color based on if the chat is transparent or not
+		String message = event.getMessage();
+		if (isChatboxTransparent() && client.getGameState().equals(GameState.LOGGED_IN))
+		{
+			client.addChatMessage(ChatMessageType.TRADE, "",
+					"<col=ffffff>[</col><col=9070ff>Twitch</col>] " + username + ": <col=" + Integer.toHexString(config.chatTColor().getRGB() & 0xFFFFFF) + ">" + message + "</col>", "Twitch");
+		}
+		else if (client.getGameState().equals(GameState.LOGGED_IN))
+		{
+			client.addChatMessage(ChatMessageType.TRADE, "",
+					"<col=000000>[</col><col=0000FF>Twitch</col>] " + username + ": <col=" + Integer.toHexString(config.chatOColor().getRGB() & 0xFFFFFF) + ">" + message + "</col>", "Twitch");
+		}
+	}
+	/**
+	 * User and room information update events
+	 * Triggers whenever the user's state is updated or they join a room
+	 * @param event
+	 */
+	@Handler
+	public void onUserStateEvent(UserStateEvent event)
+	{
+		String originalMessage = event.getOriginalMessage().toString();
+		badges = "";
+
+		//Finding if the user is a subscriber
+		if ((originalMessage.contains("subscriber=1") || originalMessage.contains("mod=1") || originalMessage.contains("value=broadcaster/1")) && StringUtils.containsIgnoreCase(originalMessage, "display-name=" + config.channelName()))
+		{
+			subscribed = true;
+			if (originalMessage.contains("subscriber=1"))
+			{
+				if (originalMessage.contains("subscriber/0"))
+				{
+					badges = "<img=9>";
+				}
+				else if (originalMessage.contains("subscriber/3"))
+				{
+					badges = "<img=8>";
+				}
+				else if (originalMessage.contains("subscriber/6"))
+				{
+					badges = "<img=7>";
+				}
+				else if (originalMessage.contains("subscriber/12"))
+				{
+					badges = "<img=6>";
+				}
+				else if (originalMessage.contains("subscriber/24"))
+				{
+					badges = "<img=5>";
+				}
+				else if (originalMessage.contains("subscriber/48"))
+				{
+					badges = "<img=4>";
+				}
+			}
+		}
+		else
+		{
+			subscribed = false;
+		}
+
+		//Finding if the user is a moderator or broadcaster which allows them to talk during emote-only
+		if (originalMessage.contains("mod=1") && StringUtils.containsIgnoreCase(originalMessage, "display-name=" + config.channelName()))
+		{
+			mod = true;
+			badges = "<img=26>" + badges;
+		}
+		else
+		{
+			mod = false;
+		}
+
+		if (originalMessage.contains(("value=broadcaster/1")) && StringUtils.containsIgnoreCase(originalMessage, "display-name=" + config.channelName()))
+		{
+			broadcaster = true;
+			badges = "<img=27>" + badges;
+		}
+		else
+		{
+			broadcaster = false;
+		}
+	}
+
+	/**
+	 * Triggers when the room state changes. I.E. Sub mode, slow mode, etc.
+	 * @param r
+	 */
+	@Handler
+	public void onRoomStateEvent(RoomStateEvent r)
+	{
+		String originalMessage = r.getOriginalMessage().toString();
+		//Checking if the subs-only state has changed
+		emoteOnly = originalMessage.contains("subs-only=1");
```

Comment:
I think this is supposed to be subsOnly ;)

Reply:
Lmao, whoopsie